examples/build-using-dockerfile/main.go

```diff
@@ -0,0 +1,157 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	"github.com/containerd/console"
+	"github.com/moby/buildkit/client"
+	"github.com/moby/buildkit/util/appcontext"
+	"github.com/moby/buildkit/util/appdefaults"
+	"github.com/moby/buildkit/util/progress/progressui"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"github.com/urfave/cli"
+	"golang.org/x/sync/errgroup"
+)
+
+func main() {
+	app := cli.NewApp()
+	app.Name = "build-using-dockerfile"
+	app.UsageText = `build-using-dockerfile [OPTIONS] PATH | URL | -`
+	app.Description = `
+build using Dockerfile.
+
+This command mimics behavior of "docker build" command so that people can easily get started with BuildKit.
+This command is NOT the replacement of "docker build", and should NOT be used for building production images.
+
+By default, the built image is loaded to Docker.
+`
+	dockerIncompatibleFlags := []cli.Flag{
+		cli.StringFlag{
+			Name:   "buildkit-addr",
+			Usage:  "buildkit daemon address",
+			EnvVar: "BUILDKIT_HOST",
+			Value:  appdefaults.Address,
+		},
+	}
+	app.Flags = append([]cli.Flag{
+		cli.StringSliceFlag{
+			Name:  "build-arg",
+			Usage: "Set build-time variables",
+		},
+		cli.StringFlag{
+			Name:  "file, f",
+			Usage: "Name of the Dockerfile (Default is 'PATH/Dockerfile')",
+		},
+		cli.StringFlag{
+			Name:  "tag, t",
+			Usage: "Name and optionally a tag in the 'name:tag' format",
+		},
+		cli.StringFlag{
+			Name:  "target",
+			Usage: "Set the target build stage to build.",
+		},
+	}, dockerIncompatibleFlags...)
+	app.Action = action
+	if err := app.Run(os.Args); err != nil {
+		fmt.Fprintf(os.Stderr, "error: %v\n", err)
+		os.Exit(1)
+	}
+}
+
+func action(clicontext *cli.Context) error {
+	if tag := clicontext.String("tag"); tag == "" {
```

Comment:
Why is this a required value?

Reply:
It is just required for exportation.
We can remove it later.